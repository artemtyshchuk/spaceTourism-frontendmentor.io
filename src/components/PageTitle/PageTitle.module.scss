.titleNumber {
  color: var(--color-white, #fff);
  font-family: "Barlow Condensed";
  font-size: 28px;
  font-style: normal;
  font-weight: 700;
  line-height: normal;
  letter-spacing: 4.725px;
  opacity: 0.25;
  padding-right: 28px;
}

.title {
  color: var(--color-white, #fff);
  font-feature-settings: "liga" off, "clig" off;

  /* desktop/text-preset-5 */
  font-family: "Barlow Condensed";
  font-size: var(--font-size-desktop-text-preset-5, 28px);
  font-style: normal;
  font-weight: 400;
  line-height: normal;
  letter-spacing: 4px;
}

@media (max-width: 992px) {
  //769
  .titleNumber {
    color: var(--color-white, #fff);
    font-feature-settings: "liga" off, "clig" off;
    font-family: "Barlow ";
    font-size: 20px;
    font-style: normal;
    font-weight: 700;
    line-height: normal;
    letter-spacing: 3px;
    padding-right: 24px;
  }

  .title {
    color: var(--color-white, #fff);
    font-feature-settings: "liga" off, "clig" off;

    /* tablet/text-preset-5 */
    font-family: "Barlow";
    font-size: var(--font-size-tablet-text-preset-5, 20px);
    font-style: normal;
    font-weight: 400;
    line-height: normal;
    letter-spacing: 3px;
  }
}

@media (max-width: 576px) {
  .titleNumber {
    color: var(--color-white, #fff);
    font-feature-settings: "liga" off, "clig" off;
    font-family: "Barlow Condensed";
    font-size: 16px;
    font-style: normal;
    font-weight: 700;
    line-height: normal;
    letter-spacing: 2.4px;
    opacity: 0.25;
    padding-right: 24px;
  }

  .title {
    text-align: center;
    color: var(--color-white, #fff);
    font-feature-settings: "liga" off, "clig" off;

    /* mobile/text-preset-6 */
    font-family: "Barlow Condensed";
    font-size: var(--font-size-mobile-text-preset-6, 16px);
    font-style: normal;
    font-weight: 400;
    line-height: normal;
    letter-spacing: 2.4px;
  }
}
