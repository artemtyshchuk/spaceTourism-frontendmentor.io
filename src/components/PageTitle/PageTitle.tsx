import styles from "./PageTitle.module.scss";

interface PageTitleProps {
  title: string;
  number: string;
}

export const PageTitle = ({ number, title }: PageTitleProps) => {

  return (
    <div className={styles.titleContainer} data-testid="page-title">
      <p className={styles.title}>
        <span className={styles.titleNumber}>{number}</span>
        {title}
      </p>
    </div>
  );
};
