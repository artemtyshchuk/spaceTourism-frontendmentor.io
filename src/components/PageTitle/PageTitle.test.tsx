import { render, screen } from "@testing-library/react";
import { PageTitle } from "./PageTitle";

describe("PageTitle", () => {
  it("should render PageTitle", () => {
    render(<PageTitle number="01" title="Test" />);

    const pageTitle = screen.getByTestId("page-title");

    expect(screen.getByText("Test")).toBeInTheDocument();
    expect(screen.getByText("01")).toBeInTheDocument();
    expect(pageTitle).toMatchSnapshot();
  });
});
