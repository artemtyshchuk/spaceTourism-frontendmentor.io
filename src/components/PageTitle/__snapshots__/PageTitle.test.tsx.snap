// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageTitle should render PageTitle 1`] = `
<div
  class="titleContainer"
  data-testid="page-title"
>
  <p
    class="title"
  >
    <span
      class="titleNumber"
    >
      01
    </span>
    Test
  </p>
</div>
`;
